### PR TITLE
Allow user extension of plugin-surround

### PR DESCRIPTION
**Pull request by eugenesvk**
So you could insert 【more bracket types】 and not use the inconvenient default aliases (why leave home row AND use a modifier ⇧ `'B': '}',` to insert a very common bracket)

(likely conflicts with https://github.com/NeoVintageous/NeoVintageous/pull/933 since this uses an updated config loader moved to rc which reads the existing config, will update the other one after this one is merged)

### CODE REVIEW

File: nv/rc.py [hunk 1]
```diff
@@ -113,3 +115,48 @@ def _parse_line(line: str):
         message('error detected while processing {} at line "{}":\n{}'.format(_file_name(), line.rstrip(), str(e)))
 
     return None
+
+
+from NeoVintageous.plugin import PACKAGE_NAME
+from NeoVintageous.nv.vim import INSERT, INTERNAL_NORMAL, NORMAL, OPERATOR_PENDING, REPLACE, SELECT, UNKNOWN, VISUAL, VISUAL_BLOCK, VISUAL_LINE
+
+cfgU_settings = (f'{PACKAGE_NAME}.sublime-settings')
```
Comment (gerardroche) [inline nv/rc.py:123]: Use the mains Preferences.sublime-settings for settings and the `get_setting` api consistent with other settings.

**Review by gerardroche — CHANGES_REQUESTED** [review]
Make sure CI is passing with tests passing and coding guidelines. flake8 is used for coding guidelines. See https://github.com/NeoVintageous/NeoVintageous/blob/master/.github/CONTRIBUTING.md

I'll update the contributing doc in few minutes with some better information.

It's difficult for me to review this because it has too many unrelated changes. Try to keep PRs very specific.

File: nv/rc.py (hunk 1 above)
Comment (eugenesvk) [reply nv/rc.py:123]: but that pollutes the mains (NV is already ~third of my general settings file) and requires unwieldy long names with prefixes `neovintageous_search_occ_style`

Comment (gerardroche) [reply nv/rc.py:123]: The `neovintageous` was mistake way back. The current prefix the old plugin prefix `vinstageous_`. I will probably be renaming the prefix later to something like `nv_`. See https://github.com/NeoVintageous/NeoVintageous/discussions/927.

**Comment by eugenesvk** [comment]
I'll uglify the coding style after other issues are resolved

File: nv/rc.py (hunk 1 above)
Comment (eugenesvk) [reply nv/rc.py:123]: if you're going to change this anyway, all the more reason to use a proper plugin config file

File: nv/plugin_surround.py [hunk 2]
```diff
@@ -40,6 +41,11 @@
 from NeoVintageous.nv.vim import enter_normal_mode
 from NeoVintageous.nv.vim import run_motion
 
+from NeoVintageous.nv.rc import cfgU
+
+from NeoVintageous.plugin import DEFAULT_LOG_LEVEL
+_log = logging.getLogger(__name__)
+_log.setLevel(DEFAULT_LOG_LEVEL)
```
[the remark this answers is not in the document]
Comment (eugenesvk) [reply nv/plugin_surround.py:48]: the relation is that later functions warn on errors, and you don't have logging in this module

**Comment by eugenesvk** [comment]
the commit message references the wrong issue, should be #738, not this #938

**Pull request closed without merging**